### PR TITLE
feat: support plugin-data

### PR DESCRIPTION
**Pull request by mdonnalley**
Adds support for plugin-data

@W-8236232@

### CODE REVIEW

File: src/connection.ts [hunk 1]
```diff
@@ -16,12 +25,70 @@ import {
...
+//   export interface Metric {
+//     requestPath: string;
+//     perfMetrics: AnyJson;
+//   }
+
+//   export interface Response {
+//     headers: AnyJson & { perfmetrics?: string };
+//     req: { path: string };
+//   }
+// }
```
Comment (WillieRuemmele) [inline src/connection.ts:69]: probably remove all of this

File: src/connection.ts [hunk 2]
```diff
@@ -16,12 +25,70 @@ import {
   RequestInfo,
   Tooling as JSForceTooling,
 } from 'jsforce';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
```
Comment (WillieRuemmele) [inline src/connection.ts:29]: why ts-ignore? Just no typings from jsforce? Just add a comment about why

**Review by WillieRuemmele — CHANGES_REQUESTED** [review]

File: src/connection.ts [hunk 3]
```diff
@@ -16,12 +25,70 @@ import {
...
+//     req: { path: string };
+//   }
+// }
+
+interface Response {
+  headers: AnyJson & { perfmetrics?: string };
+  req: { path: string };
+}
+
+// Unfortunately, we have to monkey patch HttpApi.prototype.request in order
```
Comment (amphro) [inline src/connection.ts:76]: I thought we talked about having this contained in the data plugin for now? Just have a `DataConnection extends Connection` class in the data plugin to do all this. For the auth info, you can always just create a new one if you need to.

**Review by amphro — CHANGES_REQUESTED** [review]

File: src/connection.ts (hunk 3 above)
Comment (mdonnalley) [reply src/connection.ts:76]: @amphro overriding the jsforce function doesn't work if I do it from plugin-data

**Review by WillieRuemmele — APPROVED** [review]

**Pull request closed without merging**